processing/src/main/java/io/druid/query/groupby/epinephelinae/ConcurrentGrouper.java

```diff
@@ -292,19 +325,413 @@ private boolean isParallelSortAvailable()
                    .collect(Collectors.toList());
   }
 
+  /**
+   * Merge dictionaries of {@link Grouper.KeySerde}s of {@link Grouper}s.  The result dictionary contains unique string
+   * string keys.
+   *
+   * @return merged dictionary if its size does not exceed max dictionary size.  Otherwise null.
+   */
+  @Nullable
+  private List<String> mergeDictionary()
+  {
+    final long maxDictionarySize = combineKeySerdeFactory.getMaxDictionarySize();
+    final Set<String> mergedDictionary = new HashSet<>();
+    long totalDictionarySize = 0L;
+
+    for (SpillingGrouper<KeyType> grouper : groupers) {
+      final List<String> dictionary = grouper.getDictionary();
+      totalDictionarySize += dictionary.stream().mapToLong(RowBasedGrouperHelper::estimateStringKeySize).sum();
+      if (totalDictionarySize > maxDictionarySize) {
+        return null;
+      }
+      mergedDictionary.addAll(dictionary);
+    }
+
+    return ImmutableList.copyOf(mergedDictionary);
+  }
+
+  /**
+   * Build a combining tree for the input iterators which combine input entries asynchronously.  This method is called
+   * when data are spilled and thus streaming combine is preferred to avoid too many disk accesses.
+   *
+   * @param sortedIterators sorted iterators
```

Comment:
If there is nothing to say about parameter maybe not mention it

Reply:
Removed.